src/app/backend/resource/deployment/rollout.go

```diff
@@ -0,0 +1,137 @@
+	if currRevision == "1" {
+		return nil, errors.New("No revision for rolling back ")
+	}
+	matchRS, err := GetReplicaSetFromDeployment(client, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+	for _, rs := range matchRS {
+		if rs.Annotations["deployment.kubernetes.io/revision"] == rolloutSpec.Revision {
+			updateDeployment := deployment.DeepCopy()
+			updateDeployment.Spec.Template.Spec = rs.Spec.Template.Spec
+			res, err := client.AppsV1().Deployments(namespace).Update(context.TODO(), updateDeployment, metaV1.UpdateOptions{})
+			if err != nil {
+				return nil, err
+			}
+			return &RolloutSpec{
+				Revision: res.Annotations["deployment.kubernetes.io/revision"],
+			}, nil
+		}
+	}
+	return nil, errors.New("There is no ReplicaSet that has the requested revision for the Deployment.")
+}
+
+// PauseDeployment is used to pause a deployment
+func PauseDeployment(client client.Interface, namespace, name string) (*v1.Deployment, error) {
+	deployment, err := client.AppsV1().Deployments(namespace).Get(context.TODO(), name, metaV1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if !deployment.Spec.Paused {
+		deployment.Spec.Paused = true
+		_, err = client.AppsV1().Deployments(namespace).Update(context.TODO(), deployment, metaV1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+		return deployment, nil
+	}
+	return nil, errors.New("The Deployment is already paused.")
+}
+
+// ResumeDeployment is used to resume a deployment
+func ResumeDeployment(client client.Interface, namespace, name string) (*v1.Deployment, error) {
+	deployment, err := client.AppsV1().Deployments(namespace).Get(context.TODO(), name, metaV1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if deployment.Spec.Paused {
+		deployment.Spec.Paused = false
+		_, err = client.AppsV1().Deployments(namespace).Update(context.TODO(), deployment, metaV1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+		return deployment, nil
+	}
+	return nil, errors.New("The deployment is already resumed.")
+}
+
+// RestartDeployment restarts a deployment in the manner of `kubectl rollout restart`.
+func RestartDeployment(client client.Interface, namespace, name string) (*RolloutSpec, error) {
+	deployment, err := client.AppsV1().Deployments(namespace).Get(context.TODO(), name, metaV1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+	res, err := client.AppsV1().Deployments(namespace).Update(context.TODO(), deployment, metaV1.UpdateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return &RolloutSpec{
+		Revision: res.Annotations["deployment.kubernetes.io/revision"],
+	}, nil
+}
+
+// GetReplicaSetFromDeployment return all replicaSet which is belong to the deployment
+func GetReplicaSetFromDeployment(client client.Interface, namespace, name string) ([]v1.ReplicaSet, error) {
```

Comment:
Shouldn't this search only for old replica sets?

Reply:
I am thinking of creating history, i.e. same as output of `kubectl rollout history deploy <resource name>`,  from all of related ReplicaSets, and displaying the history on the Deployment details view as a list, then putting the Rollback action on a line other than the current revision of history. So I'd like to get all of ReplicaSets here.